pyanaconda/input_checking.py

```diff
@@ -0,0 +1,755 @@
+        self._policy = policy
+        self._pwquality_settings = pwquality_settings_cache.get_settings_by_minlen(self._policy.minlen)
+        self._username = username
+        self._fullname = fullname
+        self._name_of_password = name_of_password
+        self._name_of_password_plural = name_of_password_plural
+
+    @property
+    def password(self):
+        """Password string to be checked.
+
+        :returns: password string for the check
+        :rtype: str
+        """
+        return self._password
+
+    @property
+    def password_confirmation(self):
+        """Content of the password confirmation field.
+
+        :returns: password confirmation string
+        :rtype: str
+        """
+        return self._password_confirmation
+
+    @property
+    def policy(self):
+        """Password quality policy.
+
+        :returns: password quality policy
+        """
+        return self._policy
+
+    @property
+    def pwquality_settings(self):
+        """Settings for libpwquality (if any).
+
+        :returns: libpwquality settings
+        :rtype: pwquality settings object or None
+        """
+        return self._pwquality_settings
+
+    @property
+    def username(self):
+        """The username for which the password is being set.
+
+        If no username is provided, "root" will be used.
+        Use username=None to disable the username check.
+
+        :returns: username corresponding to the password
+        :rtype: str or None
+        """
+        return self._username
+
+    @property
+    def fullname(self):
+        """The full name of the user for which the password is being set.
+
+        If no full name is provided, "root" will be used.
+
+        :returns: full user name corresponding to the password
+        :rtype: str or None
+        """
+        return self._fullname
+
+    @property
+    def name_of_password(self):
+        """Specifies how should the password be called in error messages.
+
+        In some cases we are checking a "password", but at other times it
+        might be a "passphrase", etc.
+
+        :returns: name of the password
+        :rtype: str or None
+        """
+        return self._name_of_password
+
+    @property
+    def name_of_password_plural(self):
+        """Specifies how should the password be called in error messages (plural form).
+
+        In some cases we are checking a "password", but at other times it
+        might be a "passphrase", etc.
+
+        :returns: name of the password
+        :rtype: str or None
+        """
+        return self._name_of_password_plural
+
+
+class CheckResult(object):
+    """Result of an input check."""
+
+    def __init__(self):
+        self._success = False
+        self._error_message = ""
+        self.error_message_changed = Signal()
+
+    @property
+    def success(self):
+        return self._success
+
+    @success.setter
+    def success(self, value):
+        self._success = value
```

Comment:
This is actually on purpose - the result object is created once per individual check when the check class is instantiated. This is because the result object has signals that fire when it's properties change.

So the result instance needs to remain the same for the lifetime of the check object, so that any callbacks connected to the signals remain in place.